### DISCUSSION TITLE
x

### DISCUSSION THREAD

**Discussion by ghost** [XTLS/Xray-core#1327] [General]

**Comment by cross-hello** [comment]
The first step is to get the ip of dns server. In your example, 
honey.com[https://honey.com] must could be queried via public dns 
server(you could observe Wireshark dns request). After then the following 
domain queries will be directed to dns server (
> honey.com[https://honey.com], in your example . )

----------------------------------------

*From: *architecturers ***@***.***>
*To: *XTLS/Xray-core ***@***.***>
*CC: *Subscribed ***@***.***>
*Date: *Nov 11, 2022 05:03:17
*Subject: *[XTLS/Xray-core] 求帮助 dns over https 过程 (Discussion #1327)

**Reply by cross-hello** [reply]
Sorry, misunderstand your meaning. Originally think may be honey.com specifying your vps.
If you don't use proxy, the request could be queried directly, as like `dig dns.google @8.8.8.8`.
If inside proxy connection, the request will be sent to server to process.

**Comment by cross-hello** [comment]
What you mean actually may be the first query of domain server(doh is 
abbreviation of *DNS over http*, so think may be you use to indicate the 
server) don't need doh, just a dns over tcp/udp is OK.

----------------------------------------

*From: *architecturers ***@***.***>
*To: *XTLS/Xray-core ***@***.***>
*CC: *Nanyu ***@***.***>; Comment ***@***.***>
*Date: *Nov 12, 2022 00:36:51
*Subject: *Re: [XTLS/Xray-core] 求帮助 dns over https 过程 (Discussion #1327)

**Comment by cross-hello** [comment]
😭 Didn't you even want a convention before have a communication?

----------------------------------------

*From: *architecturers ***@***.***>
*To: *XTLS/Xray-core ***@***.***>
*CC: *Nanyu ***@***.***>; Comment ***@***.***>
*Date: *Nov 12, 2022 01:24:20
*Subject: *Re: [XTLS/Xray-core] 求帮助 dns over https 过程 (Discussion #1327)

**Comment by cross-hello** [comment]
Oh, you talk that. If you have established connections with vps server, 
your domain query will be passed to the vps server.

----------------------------------------

*From: *architecturers ***@***.***>
*To: *XTLS/Xray-core ***@***.***>
*CC: *Nanyu ***@***.***>; Comment ***@***.***>
*Date: *Nov 12, 2022 02:16:29
*Subject: *Re: [XTLS/Xray-core] 求帮助 dns over https 过程 (Discussion #1327)
